packages/data/src/redux-store/index.js

```diff
@@ -259,6 +259,12 @@ export default function createReduxStore( key, options ) {
 					store,
 					resolversCache
 				);
+				privateSelectors = mapSelectorsWithResolvers(
+					privateSelectors,
+					resolvers,
+					store,
+					resolversCache
+				);
```

Comment:
great fix, and thanks for following up on this @draganescu

Reply:
Yes, this is probably the first time we're creating a private selector with a resolver :+1: However, this fix won't work when you register a private selector _after_ a store got instantiated:
```js
const store = createReduxStore( name, { ..., resolvers: { foo } );
register( store ); // instance is created and resolvers bound
unlock( store ).registerPrivateSelectors( { foo } );
```
Here you supplied a resolver for `foo`, registered a private selector `foo`, but they won't be bound together. You'd have to move the `register` call after the private registration.

That's why the private selectors are bound lazily. I'll try to do a more perfect fix in a separate PR.

Reply:
Yes, you had to move the registration, but you shouldn't have to, that's the bug 🙂 

I think this PR is mergeable with the simple resolver fix. The correct solution turns out the be a rather big PR -- although it doesn't change a lot of code, it involves a lot of refactoring.